src/vs/workbench/api/common/extHostCommands.ts

```diff
@@ -263,23 +266,26 @@ export class ExtHostCommands implements ExtHostCommandsShape {
 		}
 	}
 
-	private _reportTelemetry(command: CommandHandler, id: string) {
+	private _reportTelemetry(command: CommandHandler, id: string, duration: number) {
 		if (!command.extension || command.extension.isBuiltin) {
 			return;
 		}
 		type ExtensionActionTelemetry = {
 			extensionId: string;
 			id: string;
+			duration: number;
 		};
 		type ExtensionActionTelemetryMeta = {
 			extensionId: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The id of the extension handling the command, informing which extensions provide most-used functionality.' };
 			id: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The id of the command, to understand which specific extension features are most popular.' };
+			duration: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The duration of the command execution, to detect performance issues' };
```

Comment:
for proper classification/handling this needs to be marked as measurement, e.g [`isMeasurement: true;`](https://github.com/microsoft/vscode/blob/db0c4da8f0d760da3dae2e095b66541b08a000ad/src/vs/code/electron-main/app.ts#L1191)